src/ecpool_worker.erl

```diff
@@ -226,11 +232,29 @@ connopts([{Key, _}|Opts], Acc)
 connopts([Opt|Opts], Acc) ->
     connopts(Opts, [Opt|Acc]).
 
-reconnect(Secs, State = #state{client = Client, on_disconnect = Disconnect, supervisees = SubPids}) ->
+
+init_reconnect(#state{opts = Opts} = State, Error) ->
```

Comment:
Maybe we don't need this. The only case to return `stop` from here is when attempt limit is 1.

I'd suggest in `init/1`, it simply sends a loopback message to start the init-connect retry loop.
i.e. `init/1` always succeeds immediately so it does not slow down its supervisor.

So, the `handle_info(reconnect` clause becomes the only entry-point where connect attempts are made.

Reply:
It already does, that reconnect simply sends the "reconnect" message.

Reply:
I mean if we treat connect and reconnect equal, there is no need to handle a 'stop after first attempt' special case.

Reply:
I agree, although did that way to try and not affect any current expectant behavior. I assume ec_pool is widely used.
Ordinarily I would say if the tests pass then all should be good until something break and a new test is required

Reply:
fair.